src/engine/resources.ts

```diff
@@ -469,10 +527,28 @@ const famweightOptions: FamweightOption[] = [
 ];
 
 function planRunawayFamiliar(): RunawayFamiliarSpec {
-  const chosenFamiliar = $familiars`Frumious Bandersnatch, Pair of Stomping Boots`.find((f) =>
+  const bestFamiliar = $familiars`Frumious Bandersnatch, Pair of Stomping Boots`.find((f) =>
     have(f)
   );
+  const altFamiliar = have($familiar`Comma Chameleon`) &&
+    (getProperty("commaFamiliar") === "Frumious Bandersnatch" ||
+      getProperty("commaFamiliar") === "Pair of Stomping Boots" ||
+      getProperty("_commaRunDone"));
+
+  const chosenFamiliar = bestFamiliar !== undefined ? bestFamiliar : altFamiliar === true ? $familiar`Comma Chameleon` : false;
+
   if (chosenFamiliar) {
+
+    const commaItem = $items`aquaviolet jub-jub bird, charpuce jub-jub bird, crimsilion jub-jub bird, stomp box`.find((f) =>
+      have(f)
+    );
+
+    if (chosenFamiliar === $familiar`Comma Chameleon` && commaItem !== undefined) {
+      useFamiliar($familiar`Comma Chameleon`);
+      visitUrl(
```

Comment:
planRunawayFamiliar should not take any actual game actions. Could you rework this to use `prepare` on the entry in `getRunawaySources`?